Sources/Regex/CharacterClass.swift

```diff
@@ -242,3 +242,54 @@ extension CharacterClass: CustomStringConvertible {
     return "\(isInverted ? "not " : "")\(cc)"
   }
 }
+
+extension CharacterClass {
+  public func makeAST() -> AST? {
```

Comment:
Should it ever return nil?

Reply:
Hmm, yes, I think it could be a total function. Perhaps it makes sense to stick this with the DSL code, because it's mostly just a way to map to the "regular" (cough) AST. I'm not sure how short-term this mapping is, but in theory we could map or reconstruct custom char classes.

Given that the custom char class representation is kinda the same as the AST's, what do @natecook1000 and @hamishknight think about just storing the AST inside there and interpreting it?